### ISSUE TITLE
Codelenses now disappear and reappear with jarring jump on change

### ISSUE THREAD

**Issue by pflannery** [Microsoft/vscode#26166]
- VSCode Version: 1.12.1
- OS Version: Windows 10

Steps to Reproduce:

Its used to change smoothly

![animated-preview](https://cloud.githubusercontent.com/assets/1727302/25783896/c10af6a2-335b-11e7-869c-454f0f103604.gif)

now it jumps jarringly

![jarring-jump](https://cloud.githubusercontent.com/assets/1727302/25783887/9b1ad746-335b-11e7-8479-8918253f50b7.gif)

**Comment by jrieken** [comment]
@pflannery What extensions are those? To my knowledge we didn't change anything in the code lens world

**Comment by pflannery** [comment]
@jrieken this is the code from the extension that replaces the text https://github.com/vscode-contrib/vscode-versionlens/blob/master/src/commands/commands.js#L11

**Comment by jrieken** [comment]
Thanks, I'll take a look

**Comment by pflannery** [comment]
here's a minimal extension which replicates the issue.

[ext-codelens-jump.zip](https://github.com/Microsoft/vscode/files/984228/ext-codelens-jump.zip)

**Comment by jrieken** [comment]
The problem here is that a code lens is added for a certain range and then that range disappears (gets folded into a position). That triggers on our side to dispose that lens to ask for new lenses. @pflannery Sure you were always replacing the range of your lens? Or did the lens use an empty range before? We haven't done many changes here lately and the behaviour is kind of correct: The anchor range of a code lens is removed (replaced with some other text) which makes us retrigger this...

**Comment by pflannery** [comment]
@jrieken so far I've found that when I set the range start and end to the same position the problem seems to go away.

The only other issue I see now is where quickly replacing multiple code lenses is causing it to reload all code lenses.

**Comment by jrieken** [comment]
> The only other issue I see now is where quickly replacing multiple code lenses is causing it to reload all code lenses.

Can you share more details on that?

**Comment by roblourens** [comment]
I'm guessing that this is blocked on "needs more info" and was closed correctly, someone chime in if not. (showed up in the July verification query)
